### ISSUE TITLE
Fatal error: Low level socket error connecting to host username.webfactional.com: Connection refused (tried 1 time)

### ISSUE THREAD

**Issue by azizmb** [ashwoods/webfaction-django-fabfile#1]
Hello

When trying to deploy, im hitting an error on line #55, ie the step where the gunicorn template is uploaded. Any pointers as to what might be going wrong?

**Comment by ashwoods** [comment]
can you post your fabsettings file (with your password masked of course)

**Comment by azizmb** [comment]
```
# your projects settings

WF_HOST       =     "azizbookwala.webfactional.com"
PROJECT_NAME  =     "apnawala"
REPOSITORY    =     "git@github.com:azizmb/apnawala.git"
USER          =     "azizbookwala"
PASSWORD      =     "******"
SETTINGS_SUBDIR =   ""
VIRTUALENVS   =     "~/virtualenvs"
```

**Comment by krik** [comment]
I did some poor mans debugging and it seems this is more an issue with line #51: response = _webfaction_create_app(env.project)

Maybe that's an issue with WebFaction? For me it works 1 out of 10 times. For some reason everything's fine when setting up supervisor.

**Comment by azizmb** [comment]
Same as @krik , supervisor setup works fine, but creating the actual application fails.

**Comment by bianchimro** [comment]
Confirming the same issue reported by @krik ad @azizmb.

**Comment by ashwoods** [comment]
was testing this today. i created and recreated my site like 10 times. does it happen when the application already exists? or completely pristine?

**Comment by bianchimro** [comment]
Hi,
When I tried It was not working in any case, I made several trials with no success.

Il giorno 30/mar/2012, alle ore 19:27, Ashley Cambareply@reply.github.com ha scritto:

**Comment by azizmb** [comment]
Hey @ashwoods it happens in both cases. Ive come to figure out this is a problem with fabric, not with the script itself, downgrading to fabric 1.3.4 seemed to work

**Comment by elpargo** [comment]
Hi, I'm having the same problem

I can confirm the 1.3.\* brand does not exposes this problem.

**Comment by elpargo** [comment]
After some debugging I realized I did something very stupid. The good news is that you did as well :)

See https://github.com/ashwoods/webfaction-django-fabfile/blob/master/fabfile.py#L53

you have 
    env.port = response['port']

remember env is a global variable and it will stay the same between command later on when that code calls

https://github.com/ashwoods/webfaction-django-fabfile/blob/master/fabfile.py#L128

that run will create a new ssh session BUT that session will try to connect to webfaction based on env.port and you don't have ssh running on your app port do you :) ?

In a nutshell, change the env.port to something like like env.app_port

**Comment by ashwoods** [comment]
hey thx! the inconsistency might come from the buggy port implementation that was fixed around 10 months ago. Don't you want to have the honors for the patch? :)

**Comment by Renmiri** [comment]
apparently is a problem with ssh service not running
